nifi-nar-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/ListBoxFiles.java

```diff
@@ -0,0 +1,262 @@
+    @WritesAttribute(attribute = BoxFileInfo.TIMESTAMP, description = "The last modified time of the file.")})
+@Stateful(scopes = {Scope.CLUSTER}, description = "The processor stores necessary data to be able to keep track what files have been listed already." +
+    " What exactly needs to be stored depends on the 'Listing Strategy'.")
+public class ListBoxFiles extends AbstractListProcessor<BoxFileInfo> {
+    public static final PropertyDescriptor FOLDER_ID = new PropertyDescriptor.Builder()
+        .name("box-folder-id")
+        .displayName("Folder ID")
+        .description("The ID of the folder from which to pull list of files.")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+        .required(true)
+        .build();
+
+    public static final PropertyDescriptor RECURSIVE_SEARCH = new PropertyDescriptor.Builder()
+        .name("recursive-search")
+        .displayName("Search Recursively")
+        .description("When 'true', will include list of files from sub-folders." +
+            " Otherwise, will return only files that are within the folder defined by the 'Folder ID' property.")
+        .required(true)
+        .defaultValue("true")
+        .allowableValues("true", "false")
+        .build();
+
+    public static final PropertyDescriptor MIN_AGE = new PropertyDescriptor.Builder()
+        .name("min-age")
+        .displayName("Minimum File Age")
+        .description("The minimum age a file must be in order to be considered; any files younger than this will be ignored.")
+        .required(true)
+        .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+        .defaultValue("0 sec")
+        .build();
+
+    public static final PropertyDescriptor LISTING_STRATEGY = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(AbstractListProcessor.LISTING_STRATEGY)
+        .allowableValues(BY_TIMESTAMPS, BY_ENTITIES, BY_TIME_WINDOW, NO_TRACKING)
+        .build();
+
+    public static final PropertyDescriptor TRACKING_STATE_CACHE = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(ListedEntityTracker.TRACKING_STATE_CACHE)
+        .dependsOn(LISTING_STRATEGY, BY_ENTITIES)
+        .build();
+
+    public static final PropertyDescriptor TRACKING_TIME_WINDOW = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(ListedEntityTracker.TRACKING_TIME_WINDOW)
+        .dependsOn(LISTING_STRATEGY, BY_ENTITIES)
+        .build();
+
+    public static final PropertyDescriptor INITIAL_LISTING_TARGET = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(ListedEntityTracker.INITIAL_LISTING_TARGET)
+        .dependsOn(LISTING_STRATEGY, BY_ENTITIES)
+        .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(
+        FOLDER_ID,
+        BoxClientService.BOX_CLIENT_SERVICE,
+        RECURSIVE_SEARCH,
```

Comment:
Please move the `Box Client Service` to the first position (similar to GoogleDrive and Dropbox processors).
```suggestion
        BoxClientService.BOX_CLIENT_SERVICE,
        FOLDER_ID,
        RECURSIVE_SEARCH,
```